pkg/cache/scheduler_callback.go

```diff
@@ -60,8 +60,15 @@ func (callback *AsyncRMCallback) UpdateAllocation(response *si.AllocationRespons
 			return err
 		}
 		if app := callback.context.GetApplication(alloc.ApplicationID); app != nil {
-			ev := NewAllocateTaskEvent(app.GetApplicationID(), alloc.AllocationKey, alloc.AllocationID, alloc.NodeID)
-			dispatcher.Dispatch(ev)
+			if task := callback.context.getTask(app.GetApplicationID(), alloc.AllocationKey); task != nil {
```

Comment:
Can we get `task` from `app`? for instance: `app.GetTask(alloc.AllocationKey)`

Reply:
Either method works fine and performance isn't a concern here as this happens at most once per running task which was recovered during startup. My preference would be to leave it as-is.